crates/runtime/src/management/mod.rs

```diff
@@ -0,0 +1,372 @@
+};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Missing required secret: {name}. Specify a value."))]
+    MissingRequiredSecret { name: String },
+
+    #[snafu(display("Table provider does not support read_write mode"))]
+    NoReadWriteProvider {},
+
+    #[snafu(display(
+        "Unable to create data connector: {source}\nReport a bug to request support: https://github.com/spiceai/spiceai/issues"
+    ))]
+    UnableToCreateDataConnector {
+        source: Box<dyn std::error::Error + Sync + Send>,
+    },
+
+    #[snafu(display("{source}"))]
+    UnableToCreateCloudTableProvider { source: DataConnectorError },
+
+    #[snafu(display("Error exporting task_history records: {source}"))]
+    UnableToExportTaskHistoryData {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+}
+
+pub(crate) async fn init_cloud_management(
+    runtime: Arc<Runtime>,
+    config: &spicepod_management,
+) -> Result<(), Error> {
+    if !config.enabled {
+        return Ok(());
+    }
+
+    match Management::try_from(config, runtime).await {
+        Ok(management) => management.start().await,
+        Err(e) => Err(e),
+    }
+}
+
+/// Spice Cloud Management
+///
+/// Implements runtime management and observability for Spice instances running
+/// outside of the Spice Cloud Platform, enabling monitoring and troubleshooting capabilities
+/// across on-premises and cluster environments.
+pub(crate) struct Management {
+    runtime: Arc<Runtime>,
+    api_key: String,
+    params: HashMap<String, SecretString>,
+}
+
+impl Management {
+    pub async fn try_from(
+        config: &spicepod_management,
+        runtime: Arc<Runtime>,
+    ) -> Result<Self, Error> {
+        let params = get_params_with_secrets(runtime.secrets(), &config.params).await;
+
+        let api_key = params
+            .get("api_key")
+            .map(SecretBox::expose_secret)
+            .context(MissingRequiredSecretSnafu { name: "api_key" })?;
+
+        Ok(Self {
+            api_key: api_key.to_string(),
+            runtime,
+            params,
+        })
+    }
+
+    pub async fn start(&self) -> Result<(), Error> {
+        self.start_task_history_export().await?;
+        tracing::info!("Initialized Spice Cloud management");
+        Ok(())
+    }
+
+    async fn start_task_history_export(&self) -> Result<(), Error> {
+        let app_ref = self.runtime.app();
+        let app_lock = app_ref.read().await;
+        if let Some(app) = app_lock.as_ref() {
+            if !app.runtime.task_history.enabled {
+                tracing::debug!("Task history is disabled via configuration.");
+                return Ok(());
+            }
+        }
+        drop(app_lock);
+
+        self.init_task_history_sink_table().await?;
+
+        let cancellation_token = CancellationToken::new();
+        let df = self.runtime.datafusion();
+
+        let _task = self
+            .runtime
+            .start_runtime_task(
```

Comment:
Would this be a good fit for this? https://github.com/spiceai/spiceai/blob/trunk/crates/runtime/src/cancellable_task.rs

Reply:
Yeah 👍 - `start_runtime_task` is based on `spawn_cancellable_task` functionality with the extra logic to automatically call cancellation when runtime is shutting down. This way spawned here task is gracefully shutdown including finalization/cancellation signal that is used to perform final export